addons/repair/tests/test_repair.py

```diff
@@ -490,3 +490,86 @@ def test_repair_with_product_in_package(self):
+        self.env['stock.quant']._update_available_quantity(self.product_a, stock_location_shelf_2, 5, lot_id=lot_B)
+        self.env['stock.quant']._update_available_quantity(self.product_a, stock_location_Inventory_adjustment, 5, lot_id=lot_C)
+
+        # create a return picking
+        repair_form = Form(self.env['repair.order'])
+        repair_form.product_id = self.product_a
+        repair_form.lot_id = lot_A
+        self.assertEqual(repair_form.location_id, self.stock_warehouse.lot_stock_id)
+
+        # Check location of repair order has been updated correctly when selected lot have multiple location.
+        repair_form.lot_id = lot_B
+        self.assertEqual(repair_form.location_id, stock_location_shelf_1)
+
+        # Check when selected lot's location is not in ['internal', 'customer', 'transit']
+        repair_form.lot_id = lot_C
+        self.assertEqual(repair_form.location_id.id, False)
+
+        # Check when selected lot don't have any quant
+        repair_form.lot_id = lot_D
+        warehouse = self.env['stock.warehouse'].search([('company_id', '=', repair_form.company_id.id)], limit=1)
+        self.assertEqual(repair_form.location_id, warehouse.lot_stock_id)
+
+    def test_repair_check_picking(self):
+        # Test that when return is assign to repair order then the location will take from return's destination location
+        self.product_a.type = 'product'
+        self.env['stock.quant']._update_available_quantity(self.product_a, self.stock_warehouse.lot_stock_id, 10)
+
+        # Create new picking: 5 product_a
+        picking = Form(self.env['stock.picking'])
+        ptout = self.env['stock.picking.type'].search([('code', '=', 'outgoing')], limit=1)
+        picking.picking_type_id = ptout
+        with picking.move_ids_without_package.new() as move:
+            move.product_id = self.product_a
+            move.product_uom_qty = 5
+            move.quantity_done = 5
+        picking = picking.save()
+        picking.action_confirm()
+        picking.button_validate()
+
+        # create a return picking
+        return_wizard = self.env['stock.return.picking'].with_context(active_id=picking.id, active_ids=picking.ids).create({
+            'location_id': picking.location_id.id,
+            'picking_id': picking.id,
+        })
+        return_wizard._onchange_picking_id()
+        res = return_wizard.create_returns()
+        return_picking = self.env['stock.picking'].browse(res["res_id"])
+        return_picking.action_confirm()
+        return_picking.button_validate()
+
+        # create repair from return
+        res_dict = return_picking.action_repair_return()
+        repair_form = Form(self.env[(res_dict.get('res_model'))].with_context(res_dict['context']))
+        # Check that the location in repair is set destination location of returns
+        self.assertEqual(repair_form.location_id, return_picking.location_dest_id)
```

Comment:
if everything was done correctly, then the location should be the one set on the picking at the start, we should test that
```suggestion
        self.assertEqual(repair_form.location_id, stock_location_shelf_1)
```